src/software/thunderscope/common/common_widgets.py

```diff
@@ -380,30 +380,46 @@ def set_table_data(
     table.setHorizontalHeaderLabels(horizontal_headers)
 
 
-def change_button_state(button, enable):
-    """Change button color and clickable state.
+def disconnect_signal(signal: QtCore.Signal):
+    """Helper function to disconnect all connections for a Qt signal.
+    Suppresses TypeErrors thrown by Signal.disconnect() if there are no connections.
```

Comment:
huh weird that this is the only way we have to disconnect the signal

Reply:
Qt sucks